### PR TITLE
Optimisation at time

### LINKED ISSUE #227: Optimisation of time-varying fields
Need to be able to perform optimisation at a particular time for a time-varying field so the client can optimise over a time sequence.

### PR DESCRIPTION
**Pull request by rchristie**
Closes #227.

### CODE REVIEW

File: src/api/opencmiss/zinc/fieldparameters.h [hunk 1]
```diff
@@ -184,6 +184,26 @@ ZINC_API int cmzn_fieldparameters_set_parameters(
...
+ZINC_API double cmzn_fieldparameters_get_time(
+	cmzn_fieldparameters_id fieldparameters);
+
+/**
+ * Set time at which parameters are to be used. Default is 0.0.
+ * Client should get number of parameters after this call in case different.
+ *
+ * @param fieldparameters  The field parameters to modify.
+ * @param timeIn  The time to set. Must match a time held at nodes.
+ * @return  Result OK on success, otherwise any other error code.
```
Comment (hsorby) [inline src/api/opencmiss/zinc/fieldparameters.h]: I sort of feel that we know the possible error codes that could be returned here.  I think we should state what error codes can be returned if things didn't go well.

Comment (rchristie) [reply src/api/opencmiss/zinc/fieldparameters.h]: I don't quite yet -- in future I'm thinking it should be able to deny times which do not match a known time sequence. The only other possible fail is ERROR_ARGUMENT if fieldparameters is invalid.

File: tests/fieldmodule/fieldparameterstests.cpp [hunk 2]
```diff
@@ -47,6 +47,10 @@ TEST(ZincFieldparameters, validAPI)
 	EXPECT_TRUE(fieldOut.isValid());
 	EXPECT_EQ(coordinates, fieldOut);
 
+	EXPECT_DOUBLE_EQ(0.0, fieldparameters.getTime());
+	EXPECT_EQ(RESULT_OK, fieldparameters.setTime(1.234));
```
Comment (hsorby) [inline tests/fieldmodule/fieldparameterstests.cpp:51]: In the documentation we say:
` * @param timeIn  The time to set. Must match a time held at nodes.`
Is the time `1.234` a time held at the nodes for the cube used here?

Comment (rchristie) [reply tests/fieldmodule/fieldparameterstests.cpp:51]: No it doesn't; I'm just testing the call as simply as I can knowing it isn't currently checked.

Comment (hsorby) [reply tests/fieldmodule/fieldparameterstests.cpp:51]: I think we should change to documentation from ` Must match a time held at nodes.` too `In the future, timeIn will be required to match a time held at nodes.`

**Review by hsorby — APPROVED** [review]

**Pull request merged**